compiler/rustc_hir/src/hir.rs

```diff
@@ -3005,6 +3047,127 @@ impl Item<'_> {
     pub fn item_id(&self) -> ItemId {
         ItemId { owner_id: self.owner_id }
     }
+
+    /// Expect an [`ItemKind::ExternCrate`] or panic.
+    #[track_caller]
+    pub fn expect_extern_crate(&self) -> Option<Symbol> {
+        let ItemKind::ExternCrate(s) = self.kind else { unreachable!() };
```

Comment:
these should probably be `bug!`s

Reply:
even more ideally, they'd say "expected an extern crate, got `{:?}`" or something -- makes it much easier at a glance (or a stack trace submitted on an issue) to know what's up

Reply:
I don't think `bug!`s are available at this point in the crate graph :(

Reply:
nooooooo

ok well can u at least improve the panic message?

Reply:
Yes, I'll improve the messages (not today tho, I'm tired and tired)